test/server.test.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const Fastify = require('..')
+
+test('listen should accept null port', t => {
```

Comment:
is it should accept or should throw?

Reply:
I think we should allow since we want 'app.listen()' to work. The default behavior is to use 0 as the port if its not a valid one. I'm just validating that this is the intended behavior right now.

Reply:
I got yea before it was actually throwing an error?

Also I think if we were to change "returns error" vs "throw" is better.

Reply:
I'll need to get back to you on previous behavior. Thanks for the feedback.

Reply:
`fastify.listen()` never throw an error for the `null` or `undefined` values. My prior pull request did not change this. Here's the link to the particular change in my previous pull request: https://github.com/fastify/fastify/pull/4289/files#diff-1b48bb429de959672dadab9c87c215e19177dd70e3bd932d131ce74294c44cd5L334